plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergGcsConnectorSmokeTest.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.io.Resources;
+import io.airlift.log.Logger;
+import io.trino.hadoop.ConfigurationInstantiator;
+import io.trino.plugin.hive.containers.HiveHadoop;
+import io.trino.plugin.hive.gcs.GoogleGcsConfigurationInitializer;
+import io.trino.plugin.hive.gcs.HiveGcsConfig;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.TestingConnectorBehavior;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.iceberg.FileFormat;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Parameters;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.FileAttribute;
+import java.nio.file.attribute.PosixFilePermissions;
+import java.util.Base64;
+
+import static io.trino.plugin.hive.containers.HiveHadoop.HIVE3_IMAGE;
+import static io.trino.testing.sql.TestTable.randomTableSuffix;
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Objects.requireNonNull;
+
+public class TestIcebergGcsConnectorSmokeTest
+    extends BaseIcebergConnectorSmokeTest
+{
+    private static final Logger LOG = Logger.get(TestIcebergGcsConnectorSmokeTest.class);
+
+    private static final FileAttribute<?> READ_ONLY_PERMISSIONS = PosixFilePermissions.asFileAttribute(PosixFilePermissions.fromString("rw-r--r--"));
+    private final String schema;
+    private final String gcpStorageBucket;
+    private final Path gcpCredentialsFile;
+    private final HiveHadoop hiveHadoop;
+    private final FileSystem fileSystem;
+
+    @Parameters({"testing.gcp-storage-bucket", "testing.gcp-credentials-key"})
+    public TestIcebergGcsConnectorSmokeTest(String gcpStorageBucket, String gcpCredentialKey)
+    {
+        super(FileFormat.PARQUET);
```

Comment:
Q Any reason we use `PARQUET` instead of `ORC` (default)?

Reply:
No reason. I will switch it to use the default